File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators=['+', '-', '*', '/']
        stack=[]

        for t in tokens:
            if t not in operators:
                stack.append(int(t))
            else:
                operand2=stack.pop()
                operand1=stack.pop()
                if t=='+':
                    ans=operand1 + operand2
                elif t=='-':
                    ans=operand1 - operand2
                elif t=='*':
                    ans=operand1 * operand2
                elif t=='/':
                    ans=int((operand1) / operand2)
                stack.append(ans)
        print(stack)
        return stack[-1]

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        var operators = new HashSet<string> { "+", "-", "*", "/" };
        var stack = new Stack<int>();

        foreach (string token in tokens) {
            if (!operators.Contains(token)) {
                stack.Push(int.Parse(token));
            } else {
                int input2 = stack.Pop();
                int input1 = stack.Pop();
                int res = 0;

                switch (token) {
                    case "+":
                        res = input1 + input2;
                        break;
                    case "-":
                        res = input1 - input2;
                        break;
                    case "*":
                        res = input1 * input2;
                        break;
                    case "/":
                        res = input1 / input2;
                        break;
                }

                stack.Push(res);
            }
        }

        return stack.Pop();
    }
}
